python/ray/autoscaler/_private/kuberay/run_autoscaler.py

```diff
@@ -51,17 +63,29 @@ def run_kuberay_autoscaler(cluster_name: str, cluster_namespace: str):
         cluster_name, cluster_namespace
     )
 
-    Monitor(
-        address=ray_address,
-        # The `autoscaling_config` arg can be a dict or a `Callable: () -> dict`.
-        # In this case, it's a callable.
-        autoscaling_config=autoscaling_config_producer,
-        monitor_ip=head_ip,
-        # Let the autoscaler process exit after it hits 5 exceptions.
-        # (See ray.autoscaler._private.constants.AUTOSCALER_MAX_NUM_FAILURES.)
-        # Kubernetes will then restart the autoscaler container.
-        retry_on_failure=False,
-    ).run()
+    # Init GCS client
+    _initialize_internal_kv(GcsClient(ray_address))
```

Comment:
Why do we need to initialize the GCS client here? Why didn't we need to initialize it before?

Reply:
this is to query GCS if using autosclaer v2. You are right, we didn't do it before since there's no alternative codepaths for v1/v2. 

I think it's fine since in both Monitor (v1, v2) we are also querying GCS and initializing the kv.

Reply:
i agree- let me see if there's way I can make it w/o changing the v1 codepath. (or at least not having a global variable)